endpoints.go

```diff
@@ -33,3 +42,312 @@ func showHealth(w http.ResponseWriter, r *http.Request) {
+
+    rows, err := db.Query(queryString, surveyRef)
+
+    if err != nil {
+        http.Error(w, "get survey query failed", http.StatusInternalServerError)
+        return
+    }
+
+    var listOfSurveys = []models.Survey{}
+
+    for rows.Next(){
+        survey := models.Survey{}
+
+        rows.Scan(
+            &survey.Reference,
+            &survey.ShortName,
+            &survey.LongName,
+            &survey.LegalBasis,
+            &survey.SurveyMode,
+        )
+
+        listOfSurveys = append(listOfSurveys, survey)
+    }
+
+    if len(listOfSurveys) == 0 {
+        re := models.NewRESTError("404", "Survey not found")
+        data, err := json.Marshal(re)
+        if err != nil {
+            http.Error(w, "Error marshaling NewRestError JSON", http.StatusInternalServerError)
+            return
+        }
+
+        w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+        w.WriteHeader(http.StatusNotFound)
+        w.Write(data)
+
+        return
+    }
+
+    data, err := json.Marshal(listOfSurveys)
+    if err != nil {
+        http.Error(w, "Failed to marshal survey JSON", http.StatusInternalServerError)
+        return
+    }
+
+    logger.Logger.Info("Successfully retrieved survey from reference")
+    w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+    w.WriteHeader(http.StatusOK)
+    w.Write(data)
+
+}
+
+//Delete survey based on given reference
+func deleteSurveyByRef (w http.ResponseWriter, r *http.Request) {
+
+    if db == nil {
+        w.WriteHeader(http.StatusInternalServerError)
+        errorString := models.Error{
+            Error: "Database connection could not be found",
+        }
+        json.NewEncoder(w).Encode(errorString)
+        return
+    }
+
+    var params = mux.Vars(r)
+
+    stmt, err := db.Prepare("DELETE FROM surveyv2.survey WHERE survey_ref = $1")
+    if err != nil {
+        http.Error(w, "SQL statement not prepared", http.StatusInternalServerError)
+        return
+    }
+
+    _, err = stmt.Exec(params["reference"])
+    if err != nil {
+        http.Error(w, "SQL statement error", http.StatusInternalServerError)
+        return
+    }
+
+    logger.Logger.Info("Successfully deleted survey")
+    w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+    w.WriteHeader(http.StatusNoContent)
+}
+
+//Update survey based on JSON request
+func updateSurveyByRef (w http.ResponseWriter, r *http.Request) {
+    if db == nil {
+        w.WriteHeader(http.StatusInternalServerError)
+        errorString := models.Error{
+            Error: "Database connection could not be found",
+        }
+        json.NewEncoder(w).Encode(errorString)
+        return
+    }
+
+    var params = mux.Vars(r)
+
+    stmt, err := db.Prepare("UPDATE surveyv2.survey SET short_name = $1, long_name = $2, legal_basis = $3, survey_mode = $4 WHERE survey_ref = $5")
```

Comment:
This should be wrapped in a transaction with rollback and have deferred closes as above. We also shouldn't require them to provide all of the attributes to successfully patch - you can see an example of this at https://github.com/ONSdigital/ras-rm-party/blob/master/respondents.go#L882